src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentManagerImpl.java

```diff
@@ -0,0 +1,389 @@
+
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // get entity set ids
+            UUID studyESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID notificationESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE, NOTIFICATION, NOTIFICATION_ES );
+            UUID partOfESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PART_OF, PART_OF_ES );
+
+            // ensure study exists
+            UUID studyEKID = Preconditions
+                    .checkNotNull( commonTasksManager.getStudyEntityKeyId( organizationId, studyId ),
+                            "study entity must exist" );
+
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi
+                    .executeFilteredEntityNeighborSearch(
+                            studyESID,
+                            new EntityNeighborsFilter(
+                                    ImmutableSet.of( studyEKID ),
+                                    java.util.Optional.of( ImmutableSet.of( notificationESID ) ),
+                                    java.util.Optional.of( ImmutableSet.of( studyESID ) ),
+                                    java.util.Optional.of( ImmutableSet.of( partOfESID ) )
+                            )
+                    );
+
+            /*
+             * studies with notifications enabled have 'ol.id' property in the corresponding
+             * associationDetails set to the value of the studyId
+             * For each study, there is can only be at most 1 notification -> partof -> study association,
+             * therefore it suffices to explore the first neighbor
+             */
+            return neighbors
+                    .values()
+                    .stream()
+                    .flatMap( Collection::stream )
+                    .anyMatch( neighbor -> neighbor.getAssociationDetails().getOrDefault( OL_ID_FQN, Set.of( "" ) )
+                            .iterator().next().toString().equals( studyId.toString() ) );
+
+        } catch ( Exception e ) {
+            String error =
+                    "failed to get notification enabled status for study " + studyId + " in org " + organizationId;
+            logger.error( error, e );
+            throw new RuntimeException( error );
+        }
+    }
+
+    @Override
+    public Map<String, UUID> getPropertyTypeIds( Set<String> propertyTypeFqns ) {
+        EdmApi edmApi;
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            edmApi = apiClient.getEdmApi();
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to load EdmApi" );
+            return ImmutableMap.of();
+        }
+
+        return propertyTypeFqns.stream().map( FullQualifiedName::new ).map( fqn -> Pair
```

Comment:
this should also be talking to EdmCacheManager and not interacting with EdmApi at all